### ISSUE TITLE
URL could not be accessed: HTTP/1.1 400 Bad Request

### ISSUE THREAD

**Issue by strarsis** [PhilippBaschke/acf-pro-installer#19]
I am suddenly unable to install/update ACF PRO:
````
$ composer update

- Installing advanced-custom-fields/advanced-custom-fields-pro (5.6.0): Downloading (connecting...)

  [Composer\Downloader\TransportException]
  The 'https://connect.advancedcustomfields.com/index.php?p=pro&a=download&t=5.6.0' URL could not be accessed: HTTP/1.1 400 Bad Request
````

composer.json:
```json
[...]
  "repositories": [
    {
      "type": "composer",
      "url": "https://wpackagist.org"
    },{
      "type": "package",
      "package": {
        "name": "advanced-custom-fields/advanced-custom-fields-pro",
        "version": "5.6.0",
        "type": "wordpress-plugin",
        "dist": {
          "type": "zip",
          "url": "https://connect.advancedcustomfields.com/index.php?p=pro&a=download"
        },
        "require": {
          "philippbaschke/acf-pro-installer": "^1.0",
          "composer/installers": "^1.0"
        }
      }
    }
  ],
[...]
````

ACF PRO key is made available in .env:
````
ACF_PRO_KEY=<key>
[...]
````

**Comment by PhilippBaschke** [comment]
Thank you for your detailed report @strarsis ☺️ 

I tried reproducing your problem but your configuration works for me. However I realized something strange. When I use a wrong key in my `.env` file, the error message contains the `k` part of the URL as well:

```
[Composer\Downloader\TransportException]                                                                                                   
The 'https://connect.advancedcustomfields.com/index.php?p=pro&a=download&t=5.6.0&k=<bad key>' URL could not be accessed: HTTP/1.1 400 Bad Request
```

Can you try to run `ACF_PRO_KEY=<key> composer update` to rule out that the problem is your `.env` file?

**Comment by strarsis** [comment]
Correct, I used some deliberately incorrect `ACF_PRO_KEY` and got the same error message.
So it may be a good idea to let this installer emit a notice/warning that the key is missing/unusual (length, pattern, etc).

**Comment by PhilippBaschke** [comment]
The key warning is a great idea! 🎉 

But does your answer mean that it worked now or is it still failing? Can you test if your key contains URL characters that might be encoded (for example [here](https://meyerweb.com/eric/tools/dencoder/))?

**Comment by strarsis** [comment]
It works now. Apparently it was an issue with environment variables between WSL/Linux.

**Comment by partounian** [comment]
I have also having this issue on my Mac machine. I'm using this with Trellis and haven't really changed anything to expect this issue. I checked the server and the pro key is still in the .env file, but it sems the ULR it tries downloading from does not append `&k=[key]`.

EDIT: Fixed by removing ACF, running composer update, pushing to remote, then adding it back in.

**Comment by mickaelperrin** [comment]
I encountered this issue today and found it was a conflict with the plugin Hirak/prestissimo.

I have submitted a simple patch #20 to resolve it.

**Comment by maciekpaprocki** [comment]
Hi. I created temporary fork to fix my builds including @mickaelperrin fix. Feel free to use it. 

I will archive it (not remove though as it would brake stuff) as soon as Phillip will come back.

**Comment by tomasruud** [comment]
I got this problem today as well, but it was because we had typed the wrong version number in the repository field.

**Comment by slackday** [comment]
have this issue on one server I deploy to and I don't know why. Running `composer clear-cache` on the server before a deploy does the job.

**Comment by blakethomp** [comment]
I got this problem today when running `composer update advanced-custom-fields/advanced-custom-fields-pro`, after updating the version #, however removing the package and requiring it again allowed it to install.

`[Composer\Downloader\TransportException]
  The 'https://connect.advancedcustomfields.com/index.php?p=pro&a=download&t=5.7.1' URL could not be accessed:
  HTTP/1.1 400 Bad Request`

**Comment by leovdw** [comment]
I had the same problem today, adding the two == at the end of the key fixed it

**Comment by markushausammann** [comment]
Same problem, none of the hacks here work.
